pkg/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -225,6 +238,12 @@ if [[ -f "{{ .pathKubeletKubeconfigReal }}" ]]; then
   fi
 fi
 
+cat << 'EOF' > "$PATH_CLOUDCONFIG_FILES"
+{{ .cloudConfigFiles }}
+EOF
+
+deleteOrphan
```

Comment:
Shouldn't this also be moved below into the `if ! diff` condition, i.e. only be executed when the cloud config changes? Why did you move it here? Probably this should only run after the units have been restarted to prevent that we delete files they depend on?

Reply:
It make sense to perform this cleanup only on cloud config changes. Thanks!